src/main/java/pl/jkap/sozzt/contract/domain/ContractFacade.java

```diff
@@ -1,55 +1,60 @@
 package pl.jkap.sozzt.contract.domain;
 
 import org.springframework.context.event.EventListener;
+import org.springframework.data.domain.PageRequest;
 import pl.jkap.sozzt.contract.dto.ContractDto;
+import pl.jkap.sozzt.contract.exception.ContractNotFoundException;
 import pl.jkap.sozzt.fileContract.event.FileUploadedSpringEvent;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
 import static java.util.Objects.requireNonNull;
 
 public class ContractFacade {
 
+    private static final int PAGE_SIZE = 5;
     private final ContractRepository contractRepository;
-    private final ContractCreator contractCreator;
+    private final ContractMapper contractMapper;
 
-    public ContractFacade(ContractRepository contractRepository, ContractCreator contractCreator) {
-        this.contractCreator = contractCreator;
+    public ContractFacade(ContractRepository contractRepository, ContractMapper contractMapper) {
+        this.contractMapper = contractMapper;
         this.contractRepository = contractRepository;
     }
 
     public ContractDto addContract(ContractDto contractDTO) {
         requireNonNull(contractDTO);
-        Contract contract = contractCreator.from(contractDTO);
-        contract = contractRepository.save(contract);
-        return contract.dto();
+        ContractEntity contractEntity = contractMapper.from(contractDTO);
+        DataInputContract dataInputContract = contractMapper.dataInputStepFrom(contractEntity);
+        return contractRepository.save(dataInputContract.toContractEntity()).dto();
```

Comment:
If I see correctly the flow is: 

1. dto
2. entity
3. domain
4. entity
5. dto

The approach is okay, I only disagree with entity in point 2. Dto should have all you need to create domain object is this true?

Reply:
It's true :)